### ISSUE TITLE
4.2 Data Flip-Flop is Wrong

### ISSUE THREAD

**Issue by Draco18s** [timlg07/TheNandGame-solutions#16]
![image](https://github.com/user-attachments/assets/e3b4a0ba-848e-4939-80bc-e19e66122753)

I had to create a custom component to match the Latch from this [repo's 4.1](https://github.com/timlg07/NandGame-Solutions/blob/master/Solutions.md#42-data-flip-flop) (which is also unstable _in this use-case_ so I fixed that too) and named it S-Latch, but is otherwise identical. In addition your solution the screenshot even *says* it isn't optimal.

Using the same number of components I can fix it, however.
![image](https://github.com/user-attachments/assets/aa5c631e-9299-4eec-9c95-287a94db066e)

I am still trying to work out the optimal solution.

**Comment by Draco18s** [comment]
Unstable usage:
![image](https://github.com/user-attachments/assets/5e76a6e8-3f6f-4c68-bd63-96edae733d39)
Whoops, accidentally borked a wire when moving things around. It's still unstable, just differently.

**Comment by timlg07** [comment]
Quite some things have changed in the game tbh. I really need to play it again I guess 😅 If you have screenshots I can directly include in the repo, you can provide them via MR :)

**Comment by Draco18s** [comment]
Second screenshot can be used direct. It's not the optimal solution, but like I said I haven't figured that out yet (but at least it is correct).

**Comment by shadow-Mas-code** [comment]
In fact, this solution is still not optimal.
Can you provide a solution that uses fewer components?
@Draco18s

**Comment by shadow-Mas-code** [comment]
@Draco18s 
However, I am still very grateful because this has indeed achieved the minimum amount of nand and is correct (I did not find the correct solution online), so it will not affect the subsequent levels.

Thank you from China!

**Comment by Draco18s** [comment]
> [@Draco18s](https://github.com/Draco18s) However, I am still very grateful because this has indeed achieved the minimum amount of nand and is correct

Yeah, it's minimum nands, what the game didn't report is minimum *components* solution. Usually minimum nands *is* the minimum components solution, though you usually get this result if you decompose a component into its individual nand gates.

**Comment by shadow-Mas-code** [comment]
Yes, at least it won't affect the subsequent levels, but my goal is to pass 100% perfectly.



At 2025-01-21 03:55:15, "Draco18s" ***@***.***> wrote:

@Draco18s However, I am still very grateful because this has indeed achieved the minimum amount of nand and is correct

Yeah, it's minimum nands, what the game didn't report is minimum components solution. Usually minimum nands is the minimum components solution, though you usually get this result if you decompose a component into its individual nand gates.
